plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +215,395 @@ in
+                end
+              '';
+          }
+          {
+            name = "Docs";
+            highlight = {
+              undercurl = true;
+              fg = "#ffffff";
+              sp = "#494d64";
+            };
+            auto_close = false;
+            matcher.__raw = # Lua
+              ''
+                function(buf)
+                  return buf.name:match('%.md') or buf.name:match('%.txt')
+                end
+              '';
+          }
+        ];
       };
-    in
-    mkIf cfg.enable {
-      extraPlugins = with pkgs.vimPlugins; [
-        cfg.package
-        nvim-web-devicons
+      indicator = {
+        style = "icon";
+        icon = "▎";
+      };
+      left_trunc_marker = "";
+      max_name_length = 18;
+      max_prefix_length = 15;
+      modified_icon = "●";
+      numbers.__raw = # Lua
+        ''
+          function(opts)
+            return string.format('%s·%s', opts.raise(opts.id), opts.lower(opts.ordinal))
+          end
+        '';
+      persist_buffer_sort = true;
+      right_trunc_marker = "";
+      separator_style = "slant";
+      show_buffer_close_icons = true;
+      show_buffer_icons = true;
+      show_close_icon = true;
+      show_tab_indicators = true;
+      sort_by = "extension";
+      tab_size = 18;
+      offsets = [
+        {
+          filetype = "neo-tree";
+          text = "File Explorer";
+          text_align = "center";
+          highlight = "Directory";
+        }
       ];
-      opts.termguicolors = true;
-      extraConfigLua = ''
-        require('bufferline').setup${helpers.toLuaObject setupOptions}
-      '';
     };
+    highlights =
+      let
+        commonBgColor = "#363a4f";
+        commonFgColor = "#1e2030";
+        commonSelectedAttrs = {
+          bg = commonBgColor;
+        };
+        selectedAttrsSet = builtins.listToAttrs (
+          map
+            (name: {
+              inherit name;
+              value = commonSelectedAttrs;
+            })
+            [
+              "buffer_selected"
+              "tab_selected"
+              "numbers_selected"
+              "close_button_selected"
+              "duplicate_selected"
+              "modified_selected"
+              "info_selected"
+              "warning_selected"
+              "error_selected"
+              "hint_selected"
+              "diagnostic_selected"
+              "info_diagnostic_selected"
+              "warning_diagnostic_selected"
+              "error_diagnostic_selected"
+              "hint_diagnostic_selected"
+            ]
+        );
+      in
+      selectedAttrsSet
+      // {
+        fill = {
+          bg = commonFgColor;
+        };
+        separator = {
+          fg = commonFgColor;
+        };
+        separator_visible = {
+          fg = commonFgColor;
+        };
+        separator_selected = {
+          bg = commonBgColor;
+          fg = commonFgColor;
+        };
+      };
```

Comment:
This feels a little overkill for an example?

If users want extensive docs they should go upstream, our example should just give a taste of how to write the config in nix syntax.

I don't mind keeping it, I just wonder if it'd be a little overwhelming for users?

Reply:
I was worried about that too, but the docs just show the generated attributes.
![image](https://github.com/user-attachments/assets/706c5d37-d1e6-494b-b5bd-7c7c8ea48ddb)

 It's just my personal highlights config to fix transparent catppuccin theme issues. I will pare it down so it's not so long, though.

Reply:
> I was worried about that too, but the docs just show the generated attributes.

Yes, we just assign the value to `example` and then the docs print it using `lib.generators.toPretty`.

If you wanted to show a function application you could use `lib.literalExpression`, but I'm assuming you would rather keep the example simple.

> I will pare it down so it's not so long, though.

Yes, I was more concerned about the size since I was already aware the function calls wouldn't show in the docs.